doc/mid-based-signalling.md

```diff
@@ -0,0 +1,276 @@
+                "2345-2345-234567-2345-2345": [{ // media group uuid
+                    "media_uuid": "aaaa-aaaa-aaaaaa-aaaa-aaaa":
+                    "purpose": "m.usermedia",
+                    "kind": "video",
+                }, {
+                    "media_uuid": "bbbb-bbbb-bbbbbb-bbbb-bbbb":
+                    "purpose": "m.usermedia",
+                    "kind": "audio",
+                },
+            },
+        }
+    },
+}
+```
+
+XXX: How flat vs deep do we want the structure to be here? I've done it quite deep here,
+organised by user ID / device ID / media group UUID, but they could also just be a flat
+list of tracks. It would be more duplication but maybe less effort to read.
+
+The expected behaviour here would be for foci to essentially maintain a structure with all
+tracks being pushed to it. This structure would probably have call IDs as a top level index,
+then look very similar to the structure of the `m.call.advertise` event. It could either keep
+a reference to the transceiver it was receiving media on in the structure itself alongside
+the media UUID, or maintain a separate map of media UUID to transceiver / peer connection such
+that the first structure could be marshalled to JSON and sent to clients as-is.
+
+On the client side, the client will essentially take the `m.call.advertise` data and save it
+almost as-is. It would probably cross-reference it against the call member state events to
+ensure that it wasn't showing feeds for any users that did not have state events indicating
+that they were in the call, although if we trust the focus and assume that conf IDs are unique
+enough to be unguessable, this may be unnecessary.
+
+In the simplest case, the client will simply iterate through this structure and add every
+media UUID it finds to an `m.call.subscribe` message.
+
+The most complex part is that when the `m.call.describe` message arrives back from the focus,
+it will have to search through the data from the `m.call.advertise` message to map the tracks
+it is now receving to the right user IDs (XXX: it could build them into a map to make this lookup
+efficient, although if we make the advertise message indexed by media UUID then it already has
+a map indexed by the correct thing...)
+
+The `select_answer` is also tweaked to be more extensible-event like although is essentially
+the same:
+
+m.select\_answer
+```
+"m.select_answer": {
+    "version": "1",
+    "conf_id": "1674732106391mz4ygIc84Q2Z6mJ5",
+    "call_id": "35657a5b793ce",
+    "party_id": "123456",
+    "selected_party_id": "678910",
+}
+```
+
+Once the client receives this, it decides what tracks it wants to receive and then sends
+a subscribe message over the data channel:
+
+m.call.subscribe
+```
+"m.call.subscribe": {
+    "seq": 1,
+    "media_uuids": {
+        "aaaa-aaaa-aaaaaa-aaaa-aaaa": {
+            "width": 1024,
+            "height": 576,
+        },
+        "bbbb-bbbb-bbbbbb-bbbb-bbbb": {},
```

Comment:
For pull-based media in the future, this is going to need to include the full path (e.g. `userId/deviceId/mediaGroupUUID/mediaUUID`)

Reply:
Wait, why would this be different?

Reply:
So that the focus knows from where to request the track in case it can't read state

Reply:
Ah right, yes, although in this case the address would also need to include the SFU that the user was publishing to? I wonder if this should look more like:

```
media: [
  {
    "media_uuid": "bbbb-bbbb-bbbbbb-bbbb-bbbb",
    "user_id": "@alice:example.org",
    "device_id": "aaaaaa",
    "sfu_user_id": "@sfu:example.org",
    "sfu_device_id": "sfusfusfu",
  }
]
```